projects/power_distribution/src/front_uv_detector.c

```diff
@@ -0,0 +1,32 @@
+#include "can_transmit.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+
+static GpioAddress s_uv_comp_pin_address = { .port = GPIO_PORT_B, .pin = 0 };
+
+static InterruptSettings s_interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,
+  .priority = INTERRUPT_PRIORITY_NORMAL,
+};
+
+static void prv_pin_interrupt_handler(const GpioAddress *address, void *context) {
+  CAN_TRANSMIT_UV_CUTOFF_NOTIFICATION();
+
+  LOG_WARN("UV lockout was triggered\n");
+}
+
+void prv_register_interrupts(void) {
+  // callback occurs when pin turns low i.e falling
+  gpio_it_register_interrupt(&s_uv_comp_pin_address, &s_interrupt_settings, INTERRUPT_EDGE_FALLING,
+                             prv_pin_interrupt_handler, NULL);
+}
+
+StatusCode front_uv_detector_init() {
+  interrupt_init();
+  status_ok_or_return(gpio_init());
+  gpio_it_init();
```

Comment:
You don't need to do this here: in a multi-module project like power distribution, important libraries like interrupts, gpio, and gpio_it are only initialized in main.c. Initializing a library twice could cause problems like wiping all GPIO pins' states. This is the reason behind the `Requires ... to be initialized` comments in headers. You can just remove these three lines.

Reply:
When I take these lines out, the tests stop succeeding (the trigger function in the test doesn't occur and I'm guessing the test is waiting for an RX/TX when no CAN message gets sent). Is there something I'm missing here?

Reply:
Ah, you still need to initialize `gpio_it` in your `setup_test`, `initialize_can_and_dependencies` doesn't do that for you.